sstables/sstables.cc

```diff
@@ -3140,6 +3140,21 @@ std::optional<large_data_stats_entry> sstable::get_large_data_stat(large_data_ty
     return std::make_optional<large_data_stats_entry>();
 }
 
+malformed_sstable_exception::malformed_sstable_exception(const sstable& sst, sstring msg, const sstring& filename)
+    : malformed_sstable_exception(fmt::format("{} in sstable {} origin={}", msg, filename, sst.get_origin()))
+{
+}
+
+[[noreturn]] void throw_malformed_sstable_exception(logging::logger& log, const sstable& sst, sstring msg, const sstring& filename) {
+    auto ex = std::make_exception_ptr(malformed_sstable_exception(sst, msg, filename));
+    log.error("{} found at {}", ex, current_backtrace());
+    std::rethrow_exception(std::move(ex));
+}
```

Comment:
Won't this cause the exception to be double-printed? at the call site and at the catch site.

Also, it makes it impossible to expect the exception. For example, importing sstables might want to catch the exception and report it to the API, not log it (poor example, but it shows that the function presupposes that the exception cannot be handled).